mitodl/ocw-hugo-themes
# demo dynamic imports

#### Any background context you want to provide?
This is a demo in response to a slack message from @ibrahimjaved12, in regards to https://github.com/mitodl/ocw-hugo-themes/pull/1071#discussion_r1098682971

@ibrahimjaved12 It seems like we have three possible approaches for splitting out the VideoJS stuff:
1. `.Page.Scratch` hugo manipulations
2. inserting a script tag like you do in current version of #1071 
3. dynamic imports

This demo PR is meant to help explore (3).

Question for you... do you have any opinions yet about which approach seems best?

#### What's this PR do?
This PR shows a demo of dynamic imports. 

**HOWEVER**: This proof of concept only works with production build, NOT dev build. I think that in order for dynamic imports to work with the dev server, some aspects of how we set up the dev server need to change.

#### How should this be manually tested?
*The filepaths below are for my system, where my repos live at `~/dev/ocw-hugo-themes`, `~/dev/ocw-hugo-projects`, and `~dev/ocw-content-rc`. Modify as necessary*. 
1. First, make sure you have an env file locally. E.g., `yarn with-env --dev --print-env ''  > .env`. A `.env` file is needed for running the productionb uild.
5. Ensure you have a local copy of course `9.40-spring-2018`. If you don't, you can just run `yarn start course` to clone it. (It's the default course).
6. Run the production build: `yarn build ~/dev/ocw-content-rc/9.40-spring-2018 ~/dev/ocw-hugo-projects/ocw-course-v2/config.yaml`
7. Serve the production build: `npx serve ~/dev/ocw-content-rc/9.40-spring-2018/dist`
8. Visit the given url.
9. In dev console, `initVideoJS()`.
10. In `ocw-hugo-themes`, `ls -lh  base-theme/dist/static_shared/js`. You should see a separate chunk called something like `240.b3db3.js` that contains just the `initVideoJS` code. *Of course, the code in this PR is just "hello world", but the idea would be for all the big videojs files to end up in that separate chunk*.
